storybookjs/storybook
# Don't allow setting `Meta of={X}` if `X` is tagged with `'autodocs'`

Telescoping on https://github.com/storybookjs/storybook/pull/20364

## What I did

Throw an error if you use `<Meta of={}` for `autodocs`-tagged CSF file.

I'm not 100% sure this should be an error while other "mistakes" are just warnings. WDYT @shilman?

https://github.com/storybookjs/storybook/blob/2f713e5ce69490e61ae13b9ee934166a83f57df4/code/lib/core-server/src/utils/StoryIndexGenerator.ts#L364-L399

## How to test

- Create a MDX file with `<Meta of={..}` an autodocs-tagged CSF File.
- Check it doesn't break in other allowed circumstances (no tag, or `autodocs=true`)
